Source/Server/Weapons/WGrenadeLauncher.cs

```diff
@@ -50,11 +50,8 @@ protected override void ShootOnce()
         // Extra Z boost
         vel.z += 0.5f;
 
-        // Move projectile somewhat forward
-        Vector3D pos = client.State.pos + Vector3D.FromActorAngle(client.AimAngle, client.AimAngleZ, PROJECTILE_OFFSET);
-
         // Spawn projectile
-        new Grenade(pos + new Vector3D(0f, 0f, PROJECTILE_Z), vel, client);
+        new Grenade(GetProjectileInitialPosition(client), vel, client);
```

Comment:
If we don't want to use `PROJECTILE_Z` anymore, let's delete that variable from all the affected weapons (and I assume from _all_ the weapons, that is?).